scripts/deploy.js

```diff
@@ -0,0 +1,72 @@
+const exec = require('child_process').execSync;
+const parseArgs = require('minimist')
+const path = require('path')
+const fs = require('fs-extra')
+const git = require('simple-git')
+const ignore = require('ignore')
+
+// Parse the command-line arguments.
+const argv = parseArgs(process.argv.slice(2))
+const srcBranch = argv._[1]
+const upstreamRepo = argv._[0]
+
+const rootDir = path.join(__dirname, '..')
+
+const config = {
+  src: {
+    repo: rootDir,
+    releaseDir: path.join(rootDir, 'deploy/src/public'),
+    branch: srcBranch,
+    dir: path.join(rootDir, 'deploy/src'),
+  },
+  dist: {
+    repo: upstreamRepo,
+    repoDir: path.join(rootDir, 'deploy/dist-src'),
+    branch: `deploy/${srcBranch}-${Math.round(new Date().getTime()/1000)}`,
+    dir: path.join(rootDir, 'deploy/dist'),
+  },
+  build: 'npm install && npm run release',
+  distignore: path.join(rootDir, 'public/.distignore')
+}
+
+// Always start with fresh source and release directories.
+fs.emptyDirSync(config.src.dir)
+fs.emptyDirSync(config.dist.dir)
+fs.emptyDirSync(config.dist.repoDir)
+
+// Checkout a fresh copy of the source branch in a new directory.
+fs.copySync(
+  path.join(config.src.repo, '.git'),
+  path.join(config.src.dir, '.git')
+)
+
+// Checkout a fresh source repository.
+git(config.src.dir)
+  .reset('hard')
+  .checkout(config.src.branch)
+
+// Checkout a fresh release repository.
+git(config.dist.repoDir)
```

Comment:
@danlouw These two git commands appear to run asynchronously so it tries to run the build steps before the repos have been fully checked out. So we need to start using callbacks or promises.